src/App.js

```diff
@@ -1,23 +1,48 @@
+import {useState, useRef} from 'react';
 import styled from 'styled-components';
 
 import MagicForm from './components/MagicForm/MagicForm';
+import MagicList from './components/MagicList/MagicList.js';
+import {CodeSpan} from './components/Stylesheet/StyledSpans.js';
 
 export default function App() {
-  //
-  function analyzeSpell(spellword) {
+  // State // for the archived Spells
+  const [savedOrders, setSavedOrders] = useState([]);
+
+  //useRef is used here to find the MagicList at the End of following saveSpellOrder-function
+  const ref = useRef();
+
+  // ANALYZE SPELL // checks the Input.value from MagicForm and sends back and to: MagicList //
+  function saveSpellOrder(spellword) {
+    //
+    const trimmedInput = spellword.trim();
+    const checkedInput = trimmedInput.toLowerCase();
+    const inputInfo = findSpellMessage(checkedInput);
+    const hasError = inputInfo.includes('ERROR'); //the error-prop is used to make specific text red, the string can tell if error has happened
+    setSavedOrders([{value: spellword, info: inputInfo, error: hasError}, ...savedOrders]);
+
+    // The following parts set the scrollingPosition inside the List to bottom on new entry:
+    const List = ref.current;
+    const topPos = List.offsetTop;
+    List.scrollTop = topPos;
+  }
+
+  // ↑ FIND SPELL MESSAGE // is called within analyzeSpell to return a fitting string //
+  function findSpellMessage(spellword) {
     if (spellword === 'cd') {
```

Comment:
My suggestion would be to define an array of objects:
```const possibleSpellWords = [{command:"cd", discription:"The command"},{command:}] ```
Then you save all the else if blocks with the following:
`possibleSpellWords.forEach(object =>{
    if(object.command === spellword){
        return object.discription;
    }
		else{return "Error..."}
});`

Reply:
Hello, i tried to include this, but now it isnt working anymore. I outsourced a function and the array, but when it is returned back into the app the value is undefined. Do you have any idea? did i smnthng wrong?